### PR TITLE
update warning for image processor loading

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Similar to #28108, there is not much we/user can do with the files on the Hub.

For this image processor loading, IMO, the attribute `image_processor_class` / `feature_extractor_class` are super unlikely specified by users (they are created automatically during saving for auto class to work). So `logger.info` is fine.

### CODE REVIEW

**Comment by ydshieh** [comment]
cc @NielsRogge

**Review by NielsRogge — APPROVED** [review]
Thanks for improving this! Will avoid the confusion here: https://discuss.huggingface.co/t/error-finding-processors-image-class-loading-based-on-pattern-matching-with-feature-extractor/31890

**Comment by amyeroberts** [comment]
Thanks for opening this @ydshieh! 

As mentioned in the internal slack channel - I'd rather we had a transition with this: at least two releases instructing the users how to update before downgrading the warnings. Unlike the text_config for CLIP, loading feature extractors for vision is  deprecated and it's necessary for users to change their configs to ensure future compatibility.

**Comment by ydshieh** [comment]
Thanks @amyeroberts Sound fair.

Do you want me to change the warning to include the instructions and the version number involved (i.e. 2 release after since the current one)?

**Comment by NielsRogge** [comment]
@amyeroberts I'm not sure it's doable to expect users to update their config? That would upset a lot of users.

We have 8000+ `preprocessor_config.json` [files on the hub](https://huggingface.co/models?pipeline_tag=image-classification&library=transformers&sort=trending) => we can't update all of these right?

File: src/transformers/models/auto/image_processing_auto.py [hunk 1]
```diff
@@ -363,16 +363,25 @@ def from_pretrained(cls, pretrained_model_name_or_path, **kwargs):
...
             if feature_extractor_class is not None:
                 logger.warning(
-                    "Could not find image processor class in the image processor config or the model config. Loading"
-                    " based on pattern matching with the model's feature extractor configuration."
+                    "Could not find image processor class in the image processor config or the model config. "
+                    f"If you are the owner of {pretrained_model_name_or_path}, please update the file "
+                    "`preprocessor_config.json` to use `image_processor_type` instead of `feature_extractor_type`. "
+                    "Otherwise, you can open a pull request on this Hub repository. "
+                    "This warning will be removed in v4.40. Currently, we try to load the image processor based on "
+                    "pattern matching with the model's feature extractor configuration."
```
Comment (ydshieh) [inline src/transformers/models/auto/image_processing_auto.py:371]: @amyeroberts let me know if this is too wordy 🙏

Comment (ydshieh) [reply src/transformers/models/auto/image_processing_auto.py:371]: I put v4.40 to give it a bit more time

Comment (amyeroberts) [inline src/transformers/models/auto/image_processing_auto.py:371]: How about this? 

```suggestion
                    "Could not find image processor auto map in the image processor or model config. Loading based on "
                    "pattern matching. Please open a PR/issue to update `preprocessor_config.json` to use "
                    "`AutoImageProcessor` instead of `AutoFeatureExtractor`. This warning will be removed in v4.40"
```

**Review by amyeroberts — APPROVED** [review]
Thanks for adding @ydshieh! 

Just a suggestion on the warning. Could you update the PR title before merging?

File: src/transformers/models/auto/image_processing_auto.py (hunk 1 above)
Comment (NielsRogge) [reply src/transformers/models/auto/image_processing_auto.py]: I would make it even shorter by saying

> Could not find `image_processor_type` in the `preprocessor_config.json` file. This file requires to update `feature_extractor_type` to `image_processor_type`. This warning will be removed in v4.40"

**Comment by ydshieh** [comment]
I have to change a bit of the warning message - if you would like to take another look.

**Review by amyeroberts — APPROVED** [review]
Thanks for iterating!
